lib/java/com/google/android/material/internal/CollapsingTextHelper.java

```diff
@@ -721,9 +721,7 @@ private void calculateBaseOffsets(boolean forceRecalculate) {
       collapsedTextWidth = 0;
     }
     final int collapsedAbsGravity =
-        GravityCompat.getAbsoluteGravity(
-            collapsedTextGravity,
-            isRtl ? ViewCompat.LAYOUT_DIRECTION_RTL : ViewCompat.LAYOUT_DIRECTION_LTR);
+        GravityCompat.getAbsoluteGravity(collapsedTextGravity, ViewCompat.getLayoutDirection(view));
```

Comment:
@pubiqq unfortunately the changes in this class cause undesirable breakages in the TextInputLayout component, which also uses CollapsingTextHelper.

Here's the Before and After:
![image](https://user-images.githubusercontent.com/1420597/183556689-2bc830b1-e4fe-4f9c-8020-9df323554861.png)

We won't be able to submit this PR as is, so hopefully you are able to reproduce the issue by adding some RTL string translations to the Text Field Catalog demo, and then you can update the PR to handle this case correctly if possible. It may be tricky though, due to the different ways that TextInputLayout and CollapsingToolbarLayout work with respect to RTL.

Reply:
Good catch! I'll see what I can do.